File: benchmarks/correctness/python/gff_features.py
"""Count features by type from GFF3. Output JSON."""
import json
import sys
from collections import Counter

counts = Counter()
total = 0
with open("data/annotations.gff") as f:
    for line in f:
        if line.startswith("#") or line.strip() == "":
            continue
        fields = line.strip().split("\t")
        if len(fields) >= 3:
            counts[fields[2]] += 1
            total += 1

sorted_counts = dict(sorted(counts.items()))
json.dump({"total_features": total, "by_type": sorted_counts}, sys.stdout, indent=2)
print()

File: benchmarks/correctness/r/gff_features.R
# Count features by type from GFF3. Output JSON.
library(jsonlite)

lines <- readLines("data/annotations.gff")
lines <- lines[!grepl("^#", lines) & nchar(trimws(lines)) > 0]
total <- length(lines)

types <- sapply(lines, function(l) strsplit(l, "\t")[[1]][3])
by_type <- as.list(sort(table(types)))
by_type <- lapply(by_type, as.integer)

cat(toJSON(list(total_features = total, by_type = by_type), auto_unbox = TRUE, pretty = TRUE))
cat("\n")
